rfcs/0112-demote-darwin-support-tier.md

```diff
@@ -0,0 +1,91 @@
+---
+feature: demote_darwin_support_tier
+start-date: 2021-10-27
+author: piegames
+co-authors: many
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Move x86_64-darwin from a Tier 2 platform to Tier 3, as described in [RFC 0046](https://github.com/NixOS/rfcs/blob/master/rfcs/0046-platform-support-tiers.md).
+
+# Motivation
+[motivation]: #motivation
+
+Darwin is constantly adding additional maintainer burden. Especially, it does not live up to the requirements:
+
+- "A lot of packages built by Hydra, full ofBorg support."
+    - There is only one ofBorg builder. It can take several hours or days for an ofBorg build to finish for Darwin, if it finishes at all.
+    - Hydra builders a greatly understaffed.
+    - If Apple follows through on [their announcement to discontinue](https://www.businessinsider.com/apple-macbook-pro-discontinued) x86-based devices, the situation won't improve due to a physical lack of hardware.
+- "Most packages work, credible ambition to reach Tier 1 coverage at some point."
+    - Tier 1 support is far out of reach.
+    - We lack expertise in the greater community to support Darwin.
+- "Some ordinary packages are channel blockers on Hydra."
+    - Due to Hydra being understaffed, this is a problem (see above).
```

Comment:
I think this duplicates the above point: `A lot of packages built by Hydra, full ofBorg support` is the tooling `x86_64-darwin` is required to have, and `Some ordinary packages are channel blockers on Hydra` is part of the impact it is allowed to have on nixpkgs.

The fact it's having the impact it would be allowed to have if it fulfilled its requirements shouldn't be a motivation for demoting it.

Reply:
Half-agree. I would call «would such a demotion improve anything for anyone» as a legitimate question to decide between «demote and see if re-promotion makes sense» vs. «create an improvement strategy and see if it fixes the issues raised». But yes, maybe it should be acknowledged as a justification of relevance of the proposed radical moves, and not as an argument why demoting would fit the guidelines per se.